groups/groups.go

```diff
@@ -185,6 +185,114 @@ func List(w http.ResponseWriter, r *http.Request) {
 	writeJsonResponse(w, bytes)
 }
 
+type ActionableInput struct {
+	InstanceCount int
+	MaxInstance   int
+	MinInstance   int
+}
+
+func Increment(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	session := handlers.GetAuthSession(ctx)
+
+	vars := mux.Vars(r)
+	identifier := vars["identifier"]
+
+	// Get the Current Group Config
+	id, err := strconv.Atoi(identifier)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	group, ok := FindGroupByID(ctx, int64(id), session.AccountID)
+	if !ok {
+		http.NotFound(w, r)
+		return
+	}
+
+	input, err := buildActionableInput(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if group.Capacity >= input.MaxInstance {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	//Set the new Capacity based on rules
+	if group.Capacity+input.InstanceCount > input.MaxInstance {
+		group.Capacity = input.MaxInstance
+	} else {
+		group.Capacity = group.Capacity + input.InstanceCount
+	}
+
+	//Update the Database and the orchestration job
+	UpdateGroup(ctx, identifier, session.AccountID, group)
+
+	err = UpdateOrchestratorJob(ctx, group)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	//Return a 202 to suggest accepted
+	w.WriteHeader(http.StatusAccepted)
+}
+
+func Decrement(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	session := handlers.GetAuthSession(ctx)
+
+	vars := mux.Vars(r)
+	identifier := vars["identifier"]
+
+	// Get the Current Group Config
+	id, err := strconv.Atoi(identifier)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	group, ok := FindGroupByID(ctx, int64(id), session.AccountID)
+	if !ok {
+		http.NotFound(w, r)
+		return
+	}
+
+	input, err := buildActionableInput(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if group.Capacity <= input.MinInstance {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	//Set the new Capacity based on rules
+	if group.Capacity-input.InstanceCount < input.MinInstance {
+		group.Capacity = input.MinInstance
+	} else {
+		group.Capacity = group.Capacity - input.InstanceCount
+	}
+
+	//Update the Database and the orchestration job
+	UpdateGroup(ctx, identifier, session.AccountID, group)
```

Comment:
How do we know if `UpdateGroup` failed here?

Reply:
Good point!

The method itself is internally handling it's own errors using `log.Fatalf` and `panic()`, which is not great and witch means that we need to do a little bit of a refactor - add error handling and return errors correctly. I will create another Pull Request to address that.